modules/component/src/componentNext.ts

```diff
@@ -420,6 +420,19 @@ export class ComponentNext<P1 extends ComponentProps> {
     )
   }
 
+  list(): iComponentNext<
```

Comment:
list should also take a `keying` function that takes in the state/props and returns a `number`.